lib/matplotlib/axes/_axes.py

```diff
@@ -5182,7 +5182,22 @@ def _pcolorargs(funcname, *args, **kw):
             return X, Y, C
 
         if len(args) == 3:
+            # Check x and y for bad data...
             X, Y, C = [np.asanyarray(a) for a in args]
+            for a in X, Y:
+                a = cbook.safe_masked_invalid(a)
+                if funcname == 'pcolormesh' and np.ma.is_masked(a):
+                    raise TypeError(
```

Comment:
I would use ValueError rather than TypeError; the problem is more a matter of the information in the container than the precise type of the container.